common/ManagementTestShared/Redesign/MockTestBase.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Core.TestFramework;
+
+namespace Azure.ResourceManager.TestFramework
+{
+    public abstract class MockTestBase<TEnvironment> : RecordedTestBase<TEnvironment> where TEnvironment : MockTestEnvironment, new()
+    {
+        public MockTestBase(bool isAsync, RecordedTestMode? mode = null) : base(isAsync, mode)
+        {
+        }
+
+        public override void StopTestRecording()
+        {
+            ValidateClientInstrumentation = false;
```

Comment:
The only purpose of this base class seems to be to diable instrumentation validation. Not worth having a class for that.

Reply:
And it turns out this overriding is also not needed.

I believe it is very possible that we could only use the MockTestEnvironment to indicate the test was a mock test.